server/portal/apps/workspace/api/views.py

```diff
@@ -218,44 +226,43 @@ def post(self, request, *args, **kwargs):
                     job_post['parameterSet']['envVariables'] = [license_var]
                 del job_post['licenseType']
 
-            # url encode inputs
-            if job_post['inputs']:
-                job_post = url_parse_inputs(job_post)
-
+            # TODOv3 need to check if execution system needs keys (https://jira.tacc.utexas.edu/browse/TV3-94)
             # Get or create application based on allocation and execution system
             apps_mgr = UserApplicationsManager(request.user)
-            app = apps_mgr.get_or_create_app(job_post['appId'], job_post['allocation'])
+            print(apps_mgr)  # TODOv3 testing workaround (to avoid flake8 error)
+            # app = apps_mgr.get_or_create_app(job_post['appId'], job_post['allocation'])
 
-            if app.exec_sys:
-                return JsonResponse({"response": {"execSys": app.exec_sys.to_dict()}})
+            # TODOv3 need to check if execution system needs keys (https://jira.tacc.utexas.edu/browse/TV3-94)
+            # code: UserApplicationsManager get_or_create_app)
+            # if app.exec_sys:
+            #     return JsonResponse({"response": {"execSys": app.exec_sys.to_dict()}})
 
-            job_post['appId'] = app.id
-            del job_post['allocation']
+            if 'parameterSet' not in job_post:
+                job_post['parameterSet'] = {}
 
             if settings.DEBUG:
                 wh_base_url = settings.WH_BASE_URL + '/webhooks/'
-                jobs_wh_url = settings.WH_BASE_URL + reverse('webhooks:jobs_wh_handler')
+                # jobs_wh_url = settings.WH_BASE_URL + reverse('webhooks:jobs_wh_handler')
             else:
                 wh_base_url = request.build_absolute_uri('/webhooks/')
-                jobs_wh_url = request.build_absolute_uri(reverse('webhooks:jobs_wh_handler'))
+                # jobs_wh_url = request.build_absolute_uri(reverse('webhooks:jobs_wh_handler'))
 
-            job_post['parameters']['_webhook_base_url'] = wh_base_url
-            job_post['notifications'] = [
-                {'url': jobs_wh_url,
-                 'event': e}
-                for e in settings.PORTAL_JOB_NOTIFICATION_STATES]
-
-            # Remove any params from job_post that are not in appDef
-            job_post['parameters'] = {param: job_post['parameters'][param]
-                                      for param in job_post['parameters']
-                                      if param in [p['id'] for p in app.parameters]}
+            if 'envVariables' not in job_post['parameterSet']:
+                job_post['parameterSet']['envVariables'] = []
+            job_post['parameterSet']['envVariables'].append({'key': '_webhook_base_url', 'value':  wh_base_url})
 
             portal_name = settings.PORTAL_NAMESPACE
             job_post['tags'] = job_post.get('tags', []).append(portal_name)
```

Comment:
failing for me locally but we think my version of tapipy could be wrong. note to self to check again.